src/storage/mod.rs

```diff
@@ -191,6 +191,108 @@ where
     }
 }
 
+/// An entry to a storage which has a component associated to the entity.
+pub struct OccupiedEntry<'a, 'b: 'a, T: 'a, D: 'a> {
+    entity: Entity,
+    storage: &'a mut Storage<'b, T, D>,
+}
+
+impl<'a, 'b, T, D> OccupiedEntry<'a, 'b, T, D>
+    where T: Component,
+          D: Deref<Target = MaskedStorage<T>>,
+{
+    /// Get a reference to the component associated with the entity.
+    pub fn get(&self) -> Option<&T> {
```

Comment:
If the entry is occupied, how could this call return `None`?

Reply:
Ah, missed that one when removing all the others.